src/Xdl.Internship.Core/Xdl.Internship.Core.DataAccess/Xdl.Internship.Core.DataAccess.MongoDB/Repositories/MongoRepository.cs

```diff
@@ -0,0 +1,103 @@
+﻿using MongoDB.Bson;
+using MongoDB.Driver;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Linq.Expressions;
+using System.Text;
+using System.Threading.Tasks;
+using Xdl.Internship.Core.DataAccess.MongoDB.Attributes;
+using Xdl.Internship.Core.DataAccess.MongoDB.Entities;
+using Xdl.Internship.Core.DataAccess.MongoDB.Settings;
+
+namespace Xdl.Internship.Core.DataAccess.MongoDB.Repositories
+{
+    public class MongoRepository<TDocument> : IMongoRepository<TDocument> where TDocument : IDocument
+    {
+        private readonly IMongoCollection<TDocument> _collection;
+
+        public MongoRepository(IMongoDbSetting settings)
+        {
+            var database = new MongoClient(settings.ConnectionString).GetDatabase(settings.DatabaseName);
+            _collection = database.GetCollection<TDocument>(GetCollectionName(typeof(TDocument)));
```

Comment:
pls use lazy getting for collection, do no initialize in ctor what is not required